lib/contentmanager.js

```diff
@@ -1029,6 +1029,38 @@ ContentManager.prototype.setupRoutes = function () {
     });
   });
 
+  // update one
+  rest.put('/content/:type/switch/:id/:switchid', function (req, res, next) {
+    var type = req.params.type;
+    var id = req.params.id;
+    var switchId = req.params.switchid;
+    var val = req.body.newLayout;
+    var switchVal = req.body.siblingLayout;
+    if ((!val || 'object' !== typeof val) && (!switchVal || 'object' !== typeof switchVal)) {
+      res.statusCode = 400;
+      return res.json({ success: false, message: 'request body was not a valid object' });
+    }
+
+    // @TODO - we need to check permissions!
+    that.update(type, { _id: id }, val, function (error, updatedDocument) {
+      if (error) {
+        res.statusCode = error instanceof ContentTypeError ? 400 : 500;
+        return res.json({ success: false, message: error.message });
+      }
+    });
+
+    that.update(type, { _id: switchId }, switchVal, function (error, updatedDocument) {
+      if (error) {
+        res.statusCode = error instanceof ContentTypeError ? 400 : 500;
+        return res.json({ success: false, message: error.message });
+      }
+    });
+
+    res.statusCode = 200;
+    return res.json({ success: true });
```

Comment:
lines 1044 to 1060 could result in 3 responses to the client, should the updates fail. Also, it's possible that the first update fails, and the second would succeed, which would be undesirable, I think. 

I think that the update in lines 1052-1057 should be moved inside the first update callback, just after line 1049, and the return on 1059-1060 moved inside that second callback if there is no error.

Also, if there is an error on the second callback, the first update should be rolled back before returning the error.